zerver/tests/test_custom_profile_data.py

```diff
@@ -942,6 +1023,84 @@ def test_default_external_account_type_field(self) -> None:
         )
         self.assert_json_error(result, "Default custom field cannot be updated.")
 
+    def test_update_with_editable_by_user(self) -> None:
+        iago = self.example_user("iago")
+        aaron = self.example_user("aaron")
+        self.login("aaron")
+
+        # Create field with editable_by_user = false
+        dummy_field_data: Dict[str, Any] = {}
+        dummy_field_data["name"] = "Dummy field"
+        dummy_field_data["field_type"] = CustomProfileField.SHORT_TEXT
+        dummy_field_data["editable_by_user"] = "false"
+        result = self.api_post(iago, "/api/v1/realm/profile_fields", info=dummy_field_data)
+        result_json = self.assert_json_success(result)
+        restricted_field_id = result_json["id"]
+
+        field_data = [
+            {
+                "id": restricted_field_id,
+                "value": "test",
+            }
+        ]
+
+        # Admins can always change their own fields
+        for field_dict in iago.profile_data():
+            if field_dict["id"] == restricted_field_id:
+                self.assertEqual(field_dict["value"], None)
```

Comment:
This loop[ pattern probably would benefit from having a little helper function so that these checks can look like `assert_restricted_field_value(iago, None)`

Reply:
Added this helper function. I think this what you meant?

```python
def assert_profile_field_value(self, user: UserProfile, field_id: int, field_value: str) -> None:
    for field_dict in user.profile_data():
        if field_dict["id"] == field_id:
            self.assertEqual(field_dict["value"], field_value)
```

Reply:
Yep!